lib/search/batch_query.rb

```diff
@@ -0,0 +1,36 @@
+module Search
+  class BatchQuery < Query
+    class TooManyQueries < Error; end
+    def run(searches_params)
+      raise(TooManyQueries, 'Maximum of 10 searches per batch') unless searches_params.count <= 10
```

Comment:
Can we add something to the commit message to indicate that this is a fairly arbitrary number and it's small because we're cautious?